service/src/test/java/bio/terra/landingzone/library/landingzones/management/ResourcesDeleteManagerTest.java

```diff
@@ -0,0 +1,171 @@
+package bio.terra.landingzone.library.landingzones.management;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.deployment.LandingZoneTagKeys;
+import bio.terra.landingzone.library.landingzones.management.deleterules.LandingZoneRuleDeleteException;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.management.exception.ManagementError;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.network.models.PrivateEndpoint;
+import com.azure.resourcemanager.network.models.PrivateEndpoints;
+import com.azure.resourcemanager.resources.models.GenericResource;
+import com.azure.resourcemanager.resources.models.GenericResources;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.Answer;
+
+@Tag("unit")
+@ExtendWith(MockitoExtension.class)
+class ResourcesDeleteManagerTest {
+  private static final String LANDING_ZONE_ID = UUID.randomUUID().toString();
+  private static final String RESOURCE_GROUP_NAME = "RESOURCE_GROUP";
+
+  @Mock private ArmManagers armManagersMock;
+  @Mock private DeleteRulesVerifier deleteRulesVerifierMock;
+
+  @Mock private AzureResourceManager azureResourceManagerMock;
+  @Mock private PrivateEndpoints privateEndpointsManagerMock;
+  @Mock private GenericResources genericResourcesManagerMock;
+
+  private ResourcesDeleteManager resourcesDeleteManager;
+
+  @BeforeEach
+  void setup() {
+    mockArmManager();
+    resourcesDeleteManager = new ResourcesDeleteManager(armManagersMock, deleteRulesVerifierMock);
+  }
+
+  @Test
+  void delete_LZResourcesNotFound() throws LandingZoneRuleDeleteException {
```

Comment:
This is a bit of a misnomer -- it's not that the resources aren't found, it's that there are no resources to delete.